src/library_uuid.js

```diff
@@ -30,11 +30,13 @@ mergeInto(LibraryManager.library, {
     var uuid = null;
 
     if (ENVIRONMENT_IS_NODE) {
+#if ENVIRONMENT_MAY_BE_NODE
```

Comment:
Is it profitable to move the `if (ENVIRONMENT_IS_NODE) {` part inside the `#if`? Our expectation is that if environment is set to not-X, then it's not expected to work if it is X, so we shouldn't need to guard against it. (As I understand it, if the environment is node, we have an empty `if` body here, so this shouldn't work anyway).

Are we able to optimize out the `if` with the JS Optimizer so it doesn't matter either way?

Reply:
Moving it outside was a little messier - then there is the if-else chain to handle (where to put `else`s, that is, if we have `if (..) { } else if (..) ` then we can't remove the first if until the else, we also need to remove the else).

The JS optimizer is not capable of doing this, however, closure is, so for really-optimized builds it shouldn't matter. Would be nice to find a clean way to make it happen without closure too, at low priority I'd say.

Reply:
How about:
```c
#if A
  if (A) {
    //...
  } else
#endif // A
#if B
  if (B) {
    //...
  } else
#endif // B
#if C
  // ...
#endif // Q
  {
    //...
  }
```

Reply:
Good idea, rewritten.